libgnucash/engine/Account.cpp

```diff
@@ -2691,6 +2691,45 @@ DxaccAccountSetCurrency (Account * acc, gnc_commodity * currency)
 /********************************************************************\
 \********************************************************************/
 
+static gpointer
+account_foreach_descendant (const Account *acc, AccountCb2 thunk,
+                            gpointer user_data, gboolean sort, gboolean can_abort)
+{
+    const AccountPrivate *priv;
+    gpointer result = nullptr;
+    GList *children;
+
+    g_return_val_if_fail (GNC_IS_ACCOUNT(acc), nullptr);
+    g_return_val_if_fail (thunk, nullptr);
+
+    priv = GET_PRIVATE(acc);
+    if (sort)
+    {
+        children = g_list_copy (priv->children);
+        children = g_list_sort (children, (GCompareFunc)xaccAccountOrder);
+    }
+    else
+        children = priv->children;
+
+    for (auto node = children; node; node = node->next)
+    {
+        auto child = static_cast<Account*>(node->data);
+        result = thunk (child, user_data);
+        if (can_abort && result) break;
+
+        result = account_foreach_descendant (child, thunk, user_data, sort, can_abort);
+        if (can_abort && result) break;
+    }
+
+    if (sort)
+        g_list_free (children);
+
+    return result;
+}
+
+/********************************************************************\
+\********************************************************************/
```

Comment:
These decorations don't add anything when the functions are short... and long functions are a code smell all on their own.